components/concept/awesomebar/src/main/java/mozilla/components/concept/awesomebar/AwesomeBar.kt

```diff
@@ -93,6 +96,27 @@ interface AwesomeBar {
             BOOKMARK,
             OPEN_TAB
         }
+
+        /**
+         * Constants for defining a shared understanding of weights.
+         */
+        object Weight {
```

Comment:
Scoring - and weights - need to somehow mesh across different providers in a sensible way, and I am not sure that "suggested weights" is the right approach here.

I don't see a way to keep components acting as "suggestion providers" conceptually separate from each other, and have them somehow produce compatible (weighted) scores. Also, these components need to be free to iterate on the way they weight search results internally (e.g. places tweaking its frecency algorithm shouldn't tie into in any way).

Another aspect here is dependency between a storage layer (for example) which provides an implementation of some kind of "search" function, and a `concept-awesomebar`. I don't think it makes sense for something like `concept-storage` to depend on `concept-awesomebar` - the `Suggestion` interface is too specific to the awesomebar itself (chips, actions, etc. don't have any relation to storage, neither can a storage implementation meaningfully fulfill an "onClicked" action). So we'll end up with a separate `SearchResult (id, url, title?, descrition?, icon?)` data class in `concept-storage` which defines a subset of a `Suggestion`.

Thinking out-loud, one way to address scoring across multiple independent providers is to let them produce any scores they please, and equalize the scores at the Awesomebar implementation layer. 

- Convert suggestions from each provider onto a common scale, e.g. [0, 100] (if we can define min/max for scores it'll be easier, but not that important since those could be figured out dynamically)
- Weight suggestions by their providers (like, anything from browsing history gets higher weight than common domain suggestions)
- Finally, sort suggestions

Reply:
I do not expect any storage to implement this. This is something for `feature-awesomebar` (not pushed yet) or other feature modules. Storage should return whatever ranked results it produces. The provider implementation (not in storage) translates that into something that is suitable for the awesomebar.

Reply:
As discussed on Slack I'll open a follow-up for implementing "provider relative ranking" + "weighting providers".